src/app/authentication/authentication.component.ts

```diff
@@ -1,58 +1,98 @@
+import { ChangeDetectorRef , Component } from '@angular/core';
 import { DomSanitizer, SafeUrl } from '@angular/platform-browser';
-
 import { AppComponent } from '../app.component';
+import { GraphService } from '../graph-service';
 import { GraphExplorerComponent } from '../GraphExplorerComponent';
 import { ScopesDialogComponent } from '../scopes-dialog/scopes-dialog.component';
 import { localLogout } from './auth';
+import { AuthService } from './auth.service';
 
 @Component({
   selector: 'authentication',
   styleUrls: ['./authentication.component.css'],
   templateUrl: './authentication.component.html',
 })
+
 export class AuthenticationComponent extends GraphExplorerComponent {
-    public authInfo = this.explorerValues.authentication;
 
-    constructor(private sanitizer: DomSanitizer) {
-        super();
-    }
+  public authInfo = this.explorerValues.authentication;
 
-    public sanitize(url: string): SafeUrl {
-        return this.sanitizer.bypassSecurityTrustUrl(url);
+  constructor(private sanitizer: DomSanitizer, private authService: AuthService, private apiService: GraphService,
+              private changeDetectorRef: ChangeDetectorRef) {
+    super();
+  }
+
+  public ngOnInit() {
+    if (this.getAuthenticationStatus() === 'authenticated') {
+      this.createUserProfile();
     }
+  }
 
-    /*
-      https://docs.microsoft.com/en-us/azure/active-directory/active-directory-v2-protocols-implicit
-     */
-    public login() {
-      const loginProperties = {
-        display: 'page',
-        response_type: 'token',
-        response_mode: 'fragment',
-        nonce: 'graph_explorer',
-        prompt: 'select_account',
-        mkt: AppComponent.Options.Language,
-        scope: AppComponent.Options.DefaultUserScopes,
-      };
-
-      hello('msft').login(loginProperties);
+  public sanitize(url: string): SafeUrl {
+    return this.sanitizer.bypassSecurityTrustUrl(url);
   }
 
-    public logout() {
+  // Https://docs.microsoft.com/en-us/azure/active-directory/active-directory-v2-protocols-implicit
+  public login() {
+    AppComponent.explorerValues.authentication.status = 'authenticating';
+    this.changeDetectorRef.detectChanges();
+
+    this.authService.login()
+      .then((user) => {
+        if (user) {
+          AppComponent.explorerValues.authentication.user = user;
+          localStorage.setItem('status', 'authenticating');
+          this.createUserProfile();
+        } else {
+          localLogout();
+        }
+      }, () => {
+        localLogout();
+      });
+    }
+  public logout() {
     localLogout();
+    this.authService.logout();
   }
 
-    public getAuthenticationStatus() {
-      return this.explorerValues.authentication.status;
+  public getAuthenticationStatus() {
+    const status = localStorage.getItem('status');
+    return status;
   }
 
-    public manageScopes() {
+  public manageScopes() {
     ScopesDialogComponent.showDialog();
   }
+
+  private createUserProfile() {
```

Comment:
I would suggest this be called "DisplayUserProfile", "RefreshUserProfile" or "ShowUserProfile".  Something that indicates we are just updating the UI and not actually creating anything.